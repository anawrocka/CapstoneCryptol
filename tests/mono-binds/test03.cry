module test03 where

// expression with a free type variable. Since mono-binds
// monomorphizes everything rather than using fancy rules to generalize
// some local binds, this test probably isn't relevant anymore
test : {a} (fin a, a >= width a) => [a]
test = foo
  where
  foo = `(a)
