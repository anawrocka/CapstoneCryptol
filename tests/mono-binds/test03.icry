:set debug=on
:set mono-binds=off
:l test03.cry
:set mono-binds=on
:l test03.cry
