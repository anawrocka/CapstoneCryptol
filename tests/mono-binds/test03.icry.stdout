Loading module Cryptol
Loading module Cryptol
Loading module test03
module test03
import Cryptol
/* Not recursive */
test03::test : {a} (fin a, a >= width a) => [a]
test03::test = \{a} (fin a, a >= width a) ->
               foo a <> <>
               where
                 /* Not recursive */
                 foo : {b} (fin b, b >= width a) => [b]
                 foo = \{b} (fin b, b >= width a) -> demote a b <> <> <>
                 
               

Loading module Cryptol
Loading module test03
module test03
import Cryptol
/* Not recursive */
test03::test : {a} (fin a, a >= width a) => [a]
test03::test = \{a} (fin a, a >= width a) ->
               foo
               where
                 /* Not recursive */
                 foo : [a]
                 foo = demote a a <> <> <>
                 
               

